File: Real-Time-Search-Workshop/javascript/lab1/lab1.js
/**
 * @fileoverview Creates a node-redis client
 * @maker Joey Whelan
 */

import { createClient, SchemaFieldTypes, VectorAlgorithms, AggregateSteps, AggregateGroupByReducers } from 'redis';
import * as dotenv from 'dotenv';
dotenv.config();

export async function lab1() {
    console.log('*** Lab 1 - Connect Client ***');
    const user = process.env.USER_NAME;
    const pwd = process.env.PASSWORD;
    const url = process.env.URL;
    const port = process.env.PORT;
    
    const client = createClient({url: `redis://${user}:${pwd}@${url}:${port}`});
    await client.connect();
    let result = await client.ping();
    console.log(result);
    return client;
}

File: Real-Time-Query-Workshop/python/code/lab1.py
# Maker: Joey Whelan
from redis import from_url
import configparser

class Lab1(object):
    def run(self):
        print('*** Lab 1 - Connect Client ***');
        config = configparser.ConfigParser()
        config.read('redis.ini')

        user = config['Redis']['USER_NAME']
        pwd = config['Redis']['PASSWORD']
        url = config['Redis']['URL']
        port = config['Redis']['PORT']
    
        client = from_url(f'redis://{user}:{pwd}@{url}:{port}')
        print(client.ping())
        return client